wideq/core.py

```diff
@@ -240,166 +345,218 @@ def parse_oauth_callback(url):
+        new_access_token = refresh_auth(self.oauth_url, self.refresh_token)
+        return Auth(
+            self.gateway,
+            self.oauth_url,
+            new_access_token,
+            self.refresh_token,
+            self.user_number,
+        )
 
-    def serialize(self) -> Dict[str, str]:
+    def dump(self):
         return {
-            'access_token': self.access_token,
-            'refresh_token': self.refresh_token,
+            "access_token": self.access_token,
+            "refresh_token": self.refresh_token,
+            "oauth_url": self.oauth_url,
+            "user_number": self.user_number,
         }
 
+    @classmethod
+    def load(cls, gateway, data):
+        return cls(
+            gateway,
+            data["oauth_url"],
+            data["access_token"],
+            data["refresh_token"],
+            data["user_number"],
+        )
+
 
 class Session(object):
-    def __init__(self, auth, session_id) -> None:
+    def __init__(self, auth, session_id=None):
         self.auth = auth
         self.session_id = session_id
 
     def post(self, path, data=None):
-        """Make a POST request to the API server.
+        """Make a POST request to the APIv1 server.
 
         This is like `lgedm_post`, but it pulls the context for the
         request from an active Session.
         """
 
-        url = urljoin(self.auth.gateway.api_root + '/', path)
-        return lgedm_post(url, data, self.auth.access_token, self.session_id)
+        url = urljoin(self.auth.gateway.api_root + "/", path)
+        return lgedm2_post(
+            url,
+            data,
+            self.auth.access_token,
+            self.auth.user_number,
+            country=self.auth.gateway.country,
+            language=self.auth.gateway.language,
+        )
+
+    def get(self, path):
+        """Make a GET request to the APIv1 server."""
+
+        url = urljoin(self.auth.gateway.api_root + "/", path)
+        return thinq2_get(
+            url,
+            self.auth.access_token,
+            self.auth.user_number,
+            country=self.auth.gateway.country,
+            language=self.auth.gateway.language,
+        )
+
+    def get2(self, path):
+        """Make a GET request to the APIv2 server."""
 
-    def get_devices(self) -> List[Dict[str, Any]]:
+        url = urljoin(self.auth.gateway.api2_root + "/", path)
+        return thinq2_get(
+            url,
+            self.auth.access_token,
+            self.auth.user_number,
+            country=self.auth.gateway.country,
+            language=self.auth.gateway.language,
+        )
+
+    def get_devices(self):
         """Get a list of devices associated with the user's account.
 
         Return a list of dicts with information about the devices.
         """
-
-        return get_list(self.post('device/deviceList'), 'item')
+        devices = self.get2("service/application/dashboard").get("item", [])
+        return as_list(devices)
```

Comment:
Any particular reason to use `as_list` here instead of the older `get_list`, which makes things easier to read?